File: Backend/FemdAPI.Core/Domains/Lecture.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace FemdAPI.Core.Domains
{

    public class Lecture
    {
        [Key]
        public Guid Id { get; protected set; }
        [Required]
        public string Name { get; protected set; }
        [Required]
        public int Number { get; protected set; }
        [Required]
        public string Description { get; protected set; }
        public string PhotoPath { get; set; }

        public WordsDictionary WordsDictionary { get; set; }

        public Lecture(string name, int number, string description)
        {
            Name = name;
            Number = number;
            Description = description;
            Id = Guid.NewGuid();
        }
    }
}

File: Front_beta/FemdBeta/src/app/Models/Lecture.ts
export class Lecture
{
  Id: string;
  LectureName: string;
  LectureNumber: number;
  Desription: string;

  constructor(id: string, lectureName: string, lectureNumber: number, description: string)
  {
    this.Id = id;
    this.LectureName = lectureName;
    this.LectureNumber = lectureNumber;
    this.Desription = description;
  }
}
